cusub_perception/detection_tree/src/detection_tree_node.cpp

```diff
@@ -6,10 +6,23 @@ using namespace cv;
 
 void DetectionTree::onInit()
 {
-    sub_name = "leviathan"; // TODO Pull from config
-    ros::NodeHandle& nh = getMTNodeHandle();
+    if (is_nodelet)
+    {
+        ros::NodeHandle& nh = getMTNodeHandle();
+        cuprint("running as a nodelet.");
+        init(nh);
+    } else
+    {
+        ros::NodeHandle nh;
+        cuprint("\033[95mNOT\033[0m running as nodelet");
+        init(nh);
+    }
+}
+
+void DetectionTree::init(ros::NodeHandle& nh)
+{
     dvector_num = 0;
-    
+    sub_name = "leviathan"; // TODO Pull from config
```

Comment:
Bad luke, bad..

Reply:
If you are feeling that lazy just ask ROS for the current namespace

Reply:
yeah lol will be fixed when we have a second sub

Reply:
damn changes requested... I'll have to fix it